hail/python/hail/expr/types.py

```diff
@@ -519,7 +520,7 @@ class tndarray(HailType):
     :class:`.NDArrayExpression`, :func:`.ndarray`
     """
 
-    @typecheck_method(element_type=hail_type, ndim=int)
+    @typecheck_method(element_type=hail_type, ndim=hail_type)
     def __init__(self, element_type, ndim):
```

Comment:
I'm a bit wary about exposing `ndim` to the user as a type, rather than a number. Can we discuss this tomorrow?

Reply:
Ya, definitely. I think it's preferable that it be internal, but it introduces a bit of complexity in that you have to be careful to chose between `nd_expr.ndim` and `nd_expr._type.ndim`.